lib/serialization/BinaryMiddleware.js

```diff
@@ -198,62 +243,42 @@ class BinaryMiddleware extends SerializerMiddleware {
 						const type = identifyNumber(thing);
 						if (type === 0 && thing >= 0 && thing <= 10) {
 							// shortcut for very small numbers
-							allocate(1);
+							allocate(I8_SIZE);
 							writeU8(thing);
 							break;
 						}
-						let n;
-						for (n = 1; n < 32 && i + n < data.length; n++) {
+						/**
+						 * amount of numbers to write
+						 * @type {number}
+						 */
+						let n = 1;
+						for (; n < 32 && i + n < data.length; n++) {
 							const item = data[i + n];
 							if (typeof item !== "number") break;
 							if (identifyNumber(item) !== type) break;
 						}
-						switch (type) {
-							case 0:
-								allocate(1 + n);
-								writeU8(I8_HEADER | (n - 1));
-								while (n > 0) {
-									currentBuffer.writeInt8(
-										/** @type {number} */ (data[i]),
-										currentPosition
-									);
-									currentPosition++;
-									n--;
-									i++;
-								}
-								break;
-							case 1:
-								allocate(1 + 4 * n);
-								writeU8(I32_HEADER | (n - 1));
-								while (n > 0) {
-									currentBuffer.writeInt32LE(
-										/** @type {number} */ (data[i]),
-										currentPosition
-									);
-									currentPosition += 4;
-									n--;
-									i++;
-								}
-								break;
-							case 2:
-								allocate(1 + 8 * n);
-								writeU8(F64_HEADER | (n - 1));
-								while (n > 0) {
-									currentBuffer.writeDoubleLE(
-										/** @type {number} */ (data[i]),
-										currentPosition
-									);
-									currentPosition += 8;
-									n--;
-									i++;
-								}
-								break;
+						const size = NUMBERS_SIZES_BY_TYPE[type];
+						const writeNumber = getWriteNumberMethod(type, currentBuffer);
+
+						allocate(HEADER_SIZE + size * n);
+						writeU8(NUMBERS_HEADERS_BY_TYPE[type] | (n - 1));
+
+						while (n > 0) {
```

Comment:
I don't think this change is a good idea. I makes `writeNumber` and `readNumber` calls polymorphic which affects performance. Before that change v8 was able to inline and optimize these loops. Now I guess that this is no longer possible (not super sure).

Anyway I feel like this part of the code should be copied for each number type to make it obvious how performance is affected.

Reply:
Should I write any explanation in code about this design?